### ISSUE TITLE
Incorrect error message in governance:propose

### ISSUE THREAD

**Issue by asaj** [celo-org/celo-monorepo#7308]
### Expected Behavior

Having an insufficient balance when making a governance proposal should fail saying that your balance is insufficient

### Current Behavior

```
I’m getting some weird behavior on alfajores where the `governance:propose` cli check for minDeposit passes but contractkit gas estimation fails and I’m super stumped
I have double checked the minDeposit a few times now…
./bin/run governance:propose --deposit 100000000000000000000 --from 0xE7818b0e067Bc205B0a2A3055818083D13F11aA8 --privateKey XXX --descriptionURL "asdf" --jsonTransactions ./proxyproposal.json
Running Checks:
   ✔  Account has at least 100 CELO
   ✔  Deposit is greater than or equal to governance proposal minDeposit
Sending Transaction: proposeTx... !
Error: Gas estimation failed: Too small deposit
    at ContractKit.<anonymous> (~/celo/monorepo/packages/contractkit/src/kit.ts:338:15)
    at step (~/celo/monorepo/packages/contractkit/lib/kit.js:44:23)
    at Object.throw (~/celo/monorepo/packages/contractkit/lib/kit.js:25:53)
    at rejected (~/celo/monorepo/packages/contractkit/lib/kit.js:17:65)
```

with a follow up message:

```
my feeCurrency was configured on disk to cUSD for which I had insufficient balance so estimation was failing but surfacing the wrong revert message
```

**Comment by asaj** [comment]
```
Not sure it matters but other methods seem to use value.toFixed()... and governance:propose uses deposit.toString() 
```

**Comment by aslawson** [comment]
Real error is that users did not have enough gas since they had cUSD configured as gascurrency. Should have this error rather than `Too small deposit`

Can also print what feecurrency is set to and perhaps add check for adequate fee currency prior to making onchain request (is this possible?)

**Comment by aslawson** [comment]
@gastonponti we have an estimategas command. is it possible to use this to add a check to the utils check builder that the individual has enough gas prior to making onchain requests?

**Comment by gastonponti** [comment]
@aslawson  Yes, we have these checks in the util: hasEnoughCelo and hasEnoughUsd that could be used here.
The only issue here could be the commands with multiple calls, and how we could handle that (probably will require to estimate all the gas first, and then trigger everything).

But seeing the actual issue, I THINK that the thing that is failing is the gas estimation call, due to that check.

So basically the error is this check https://github.com/celo-org/celo-monorepo/blob/master/packages/protocol/contracts/governance/Governance.sol#L452
but was triggered in the gas estimation call
